client/alloc_runner.go

```diff
@@ -80,19 +89,20 @@ type allocRunnerState struct {
 
 // NewAllocRunner is used to create a new allocation context
 func NewAllocRunner(logger *log.Logger, config *config.Config, updater AllocStateUpdater,
-	alloc *structs.Allocation) *AllocRunner {
+	alloc *structs.Allocation, vaultClient vaultclient.VaultClient) *AllocRunner {
 	ar := &AllocRunner{
-		config:     config,
-		updater:    updater,
-		logger:     logger,
-		alloc:      alloc,
-		dirtyCh:    make(chan struct{}, 1),
-		tasks:      make(map[string]*TaskRunner),
-		taskStates: copyTaskStates(alloc.TaskStates),
-		restored:   make(map[string]struct{}),
-		updateCh:   make(chan *structs.Allocation, 64),
-		destroyCh:  make(chan struct{}),
-		waitCh:     make(chan struct{}),
+		config:      config,
+		updater:     updater,
+		logger:      logger,
+		alloc:       alloc,
+		dirtyCh:     make(chan struct{}, 1),
+		tasks:       make(map[string]*TaskRunner),
+		taskStates:  copyTaskStates(alloc.TaskStates),
+		restored:    make(map[string]struct{}),
+		updateCh:    make(chan *structs.Allocation, 64),
+		destroyCh:   make(chan struct{}),
+		waitCh:      make(chan struct{}),
+		vaultClient: vaultClient,
```

Comment:
Yeah and it is not always need. Most likely the majority of tasks won't have vault tokens